src/main/java/com/dataloom/organization/Organization.java

```diff
@@ -29,25 +34,46 @@ public Organization(
             @JsonProperty( SerializationConstants.DESCRIPTION_FIELD ) Optional<String> description,
             @JsonProperty( SerializationConstants.EMAILS_FIELD ) Set<String> autoApprovedEmails,
             @JsonProperty( SerializationConstants.MEMBERS_FIELD ) Set<Principal> members,
-            @JsonProperty( SerializationConstants.ROLES ) Set<Role> roles ) {
-        this( new SecurablePrincipal( id, principal, title, description ), autoApprovedEmails, members, roles );
-        //        checkArgument( principal.getType().equals( PrincipalType.ORGANIZATION ) );
-        //        this.principal = new SecurablePrincipal( id, principal, title, description );
-        //        this.autoApprovedEmails = checkNotNull( autoApprovedEmails );
-        //        this.members = checkNotNull( members );
-        //        this.roles = checkNotNull( roles );
+            @JsonProperty( SerializationConstants.ROLES ) Set<Role> roles,
+            @JsonProperty( SerializationConstants.APPS ) Set<UUID> apps ) {
+        this( new SecurablePrincipal( id, principal, title, description ), autoApprovedEmails, members, roles, apps );
     }
 
     public Organization(
             SecurablePrincipal securablePrincipal,
             Set<String> autoApprovedEmails,
             Set<Principal> members,
-            Set<Role> roles ) {
+            Set<Role> roles,
+            Set<UUID> apps ) {
         checkArgument( securablePrincipal.getPrincipalType().equals( PrincipalType.ORGANIZATION ) );
         this.securablePrincipal = securablePrincipal;
         this.autoApprovedEmails = checkNotNull( autoApprovedEmails );
         this.members = checkNotNull( members );
         this.roles = checkNotNull( roles );
+        this.apps = checkNotNull( apps );
+    }
+
+    public Organization(
+            SecurablePrincipal principal,
+            Set<String> autoApprovedEmails,
+            Set<Principal> members,
+            Set<Role> roles ) {
+        this( principal, autoApprovedEmails, members, roles, ImmutableSet.of() );
+    }
+
+    public Organization(
+            Optional<UUID> id,
+            Principal principal,
+            String title,
+            Optional<String> description,
+            Set<String> autoApprovedEmails,
+            Set<Principal> members,
+            Set<Role> roles ) {
+        this( id, principal, title, description, autoApprovedEmails, members, roles, ImmutableSet.of() );
+    }
+
+    public List<UUID> getAclKey() {
+        return securablePrincipal.getAclKey();
```

Comment:
If this field is going to appear in the JSON then you may need to configure JSON Jackson to ignore missing properties in the constructor and a ```@JsonProperty``` annotation here. Otherwise you can just ```@JsonIgnore``` and it should work.